### ISSUE TITLE
Cache / --vacuum + reading from pipes bugs? Different article lists when reading an atom feed from pipes of the same content

### ISSUE THREAD

**Issue by giacecco** [akrennmair/newsbeuter#488]
Hi All, I've been struggling with this for several days, I believe I'm observing a combination of:

1) a bug in the cache, or *--vacuum* not doing its work, and
2) a bug in reading from pipes with the ```exec``` directive in the ```url``` file.

**Newsbeuter version (copy from `newsbeuter -v`)**:

```
newsbeuter 2.10-6c1d - http://www.newsbeuter.org/
Copyright (C) 2006-2015 Andreas Krennmair
Copyright (C) 2015-2017 Alexander Batischev
Copyright (C) 2006-2017 Newsbeuter contributors

newsbeuter is free software and licensed under the MIT/X Consortium License.
Type `newsbeuter -vv' for more information.

newsbeuter 2.10-6c1d
System: Linux 4.9.5-100.fc24.x86_64 (x86_64)
Compiler: g++ 6.3.1 20161221 (Red Hat 6.3.1-1)
ncurses: ncurses 6.0.20160709 (compiled with 6.0)
libcurl: libcurl/7.47.1 NSS/3.27 zlib/1.2.8 libidn/1.33 libpsl/0.13.0 (+libidn/1.33) libssh2/1.7.0 nghttp2/1.7.1 (compiled with 7.47.1)
SQLite: 3.13.0 (compiled with 3.13.0)
libxml2: compiled with 2.9.3
```

My ```config``` is [here](https://gist.github.com/f625606f25efbf0b5ec835df17dd95f4).

Steps to reproduce the issue:

1. Clear newsbeuter's cache with ```newsbeuter -X```, just to be safe
2. Write a script that generates an atom feed to stdout, suitable to be exec-ed on the fly by newsbeuter
3. Dump the stdout of (2) into a file (e.g. [here](https://gist.github.com/8ddeaa25292fbd576c39ee21d5f9ac69))
4. Configure the ```urls``` files to read both (2) and (3) (e.g. [here](https://gist.github.com/2e083447db526ff8727f8fb5247926b6))
5. Launch newsbeuter with ```-r``` and check the results. 

You would expect the two feeds to be identical but on my newsbeuter they're dramatically different. (2) has fewer and older messages, and duplicates. This is the top of the article list of (2):

![image](https://cloud.githubusercontent.com/assets/303693/22456386/1100c6aa-e78a-11e6-9e8b-fad85216bbbd.png)

This is the top of the article list of (3): 

![image](https://cloud.githubusercontent.com/assets/303693/22456408/2999cdc4-e78a-11e6-9bd4-1cd4d0a6103f.png)

My best guess is that the cache is not really been vacuumed and what I am seeing for (2) is dirt from when I was debugging the script. 

Moreover, (2) is smaller perhaps because reading from the pipe breaks at some point, and only the older articles are loaded (you can see from [my (3)](https://gist.github.com/8ddeaa25292fbd576c39ee21d5f9ac69) that the articles are listed from the older to the newer).

**BONUS** - manually delete (on my Fedora) anything in *~/.local/share/newsbeuter* and check the results of ```newsbeuter -r``` again.

This is the top of the article list of (2), where the duplicates have disappeared, but the articles are still older and fewer in number.

![image](https://cloud.githubusercontent.com/assets/303693/22456694/f293c9f4-e78b-11e6-98ec-81c191b3b05f.png)

This is the top of the article list of (3), identical to before: 

![image](https://cloud.githubusercontent.com/assets/303693/22456705/0a2335a0-e78c-11e6-91c3-4ff7029fe9f7.png)

Sorry for the long text but it wasn't trivial to debug and explain. I hope it was useful.

**Comment by Minoru** [comment]
You sure this isn't the same as #193? What happens if you only have one feed configured? Do all the articles show up in this case?

**Comment by giacecco** [comment]
Yes I'm positive. The example I give above is a simplified version of my real ```urls``` file, where all feeds have distinct names. 

If I load only one of the two I have the same issue: the ```cat```-based pipe works, the other doesn't.

**Comment by Minoru** [comment]
> The example I give above is a simplified version of my real urls file, where all feeds have distinct names.

The crux of #193 is items having the same GUIDs, not feeds having the same names. Here you're using the same feed (same items) twice, that's why I immediately thought of #193.

Also a thing I should've pointed out in the beginning, but missed:

> Clear newsbeuter's cache with newsbeuter -X, just to be safe

`-X` actually doesn't remove anything from the cache. it compacts it by re-using the empty space that was freed by, say, deleted items. The correct way to ensure that the cache is clean is to remove the cache :)

> If I load only one of the two I have the same issue: the cat-based pipe works, the other doesn't.

Do the same, but with new cache each time. If this still results in different counts, this isn't a duplicate of #193.

**Comment by giacecco** [comment]
Thanks @Minoru, I'll work on another experiment to explore what's happening here.

**Comment by Minoru** [comment]
Oh, um, I didn't mean to make you close the issue ._. I thought you'll just re-try the same thing, but *deleting* the cache this time, and will also try reloading each feed separately. For now, I just want to know if this is the same as #193 or different.

Let's reopen this and close only if it's fixed or is proven to be a duplicate of something else.
